test/extended/images/hardprune.go

```diff
@@ -177,42 +178,44 @@ var _ = g.Describe("[Feature:ImagePrune][registry][Serial][Suite:openshift/regis
 		err = AssertDeletedStorageFiles(deleted, expectedDeletions)
 		o.Expect(err).NotTo(o.HaveOccurred())
 
-		err = oc.AsAdmin().ImageClient().Image().Images().Delete(childImg1, nil)
+		err = oc.AsAdmin().ImageClient().ImageV1().Images().Delete(childImg1, nil)
 		o.Expect(err).NotTo(o.HaveOccurred())
 		// The repository a-tagged will not be removed even though it has no tags anymore.
 		// For the repository to be removed, the image stream itself needs to be deleted.
-		err = oc.AsAdmin().ImageClient().Image().ImageStreamTags(oc.Namespace()).Delete("a-tagged:latest", nil)
+		err = oc.AsAdmin().ImageClient().ImageV1().ImageStreamTags(oc.Namespace()).Delete("a-tagged:latest", nil)
 		o.Expect(err).NotTo(o.HaveOccurred())
 		deleted, err = RunHardPrune(oc, dryRun)
 		o.Expect(err).NotTo(o.HaveOccurred())
+		exutil.ImageWithMetadataOrDie(imgs[childImg1])
 		expectedDeletions = mergeOrSetExpectedDeletions(expectedDeletions,
 			&RegistryStorageFiles{
 				/* TODO: reenable once we delete layer links as well
 				LayerLinks: RepoLinks{oc.Namespace()+"/c": []string{
-					imgs[childImg1].DockerImageMetadata.ID,
+					imgs[childImg1].DockerImageMetadata.Object.(*docker10.DockerImage).ID,
```

Comment:
Nasty :stuck_out_tongue_closed_eyes:

Reply:
pre-existing

/cc @adambkaplan @dmage 

I hope you guys have BZ that track this.

Reply:
If we didn't have one before, we do now: https://bugzilla.redhat.com/show_bug.cgi?id=1718878